mdn/developer-portal
# 1438: Deprecate `image` field on pages in favour of just `card_image` (and populate data)

This changeset lays in groundwork for cleaner support of different aspect-ratio images in the CMS

**Context:**
Prior to this changeset, a Page subclass could have an Image (which would be displayed in the actual rendered page in the old design. This image is no longer shown in the page in the new design).

Additionally, there could be a "Card image", which is a smaller image used when a page is represented as a "Card" in a list of items (eg in the /posts/ list or /events/ in the redesign).

If no "Card Image" was present, the main "Page Image" was used to populate the Card's image.


* *Key changes**
Given the deprecation of a Page image, plus the desire to add support for 3:2-aspect-ratio images not just 16:9 images (which is coming in a separate PR), it makes sense to drop the `image` field from all models that feature it, and first to set the "Card image" to the value of the "Page image", so we don't lose any data and we end up populating the "Card image" with a useful value.

All code that referenced the `image` field has been updated (HTML and Python), including the automatic importer that creates `ExternalPost`s and `Video`s from other Mozilla sites


## How to test

**Note:** if you see any fields that mention "3:2 image" on dev, please ignore them - they are part of the PR that follows this one.

- Code is staged on dev and deployed ran fine. You can check in the admin to see that pages feature no `image` field now, but the _Card_ tab of each Page features a `card_image` field, which will be populated if the page originally had an image (eg see [here](https://developer-portal.dev.mdn.mozit.cloud/admin/pages/345/edit/#tab-card) and [here](https://developer-portal.dev.mdn.mozit.cloud/admin/pages/161/edit/#tab-card) )

- When reviewing code, the function that maps `foo.image` to `foo.card_image` is defined once and imported into each migration, making for less repetition and hopefully easier reviewing
